### PR TITLE
Dplan 3217 add planing document

### PR DESCRIPTION
**Pull request by sierrazamudiodemos**
### Ticket
https://demoseurope.youtrack.cloud/issue/DPLAN-3217/ADO-Issue-15115-Integration-EWM-Part-2-Stellungnahmedetailansicht-aus-EWM-und-dplan-vereinen

Add additional fields to STN json api related endpoints so that FE can show them/update/delete on the new stn details page

### How to review/test
**Attachements**
1. There are 2 types of attachements that belong to STN
- General Attachment: in this PR is represented by the resourceType GeneralStatementAttachemnt
- Source Attachment: in this PR is represented by the resourceType SourceStatementAttachment
2. The attachements can be created (POST), deleted (DELETE) and also gotten as include fields of the Statement Resource Type
3. Previously the GeneralAttachements and SourceAttachements were handled by the same resourceType StatementAttachments, Now this was devided into the 2 mentioned resource types

**PlaningDocuments**
1. documents field was enabled to list documents of a respective PlaningDocumen

### PR Checklist

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [x] Link all relevant tickets
- [ ] Move the tickets on the board accordingly
- [ ] Data-Cy attributes added/updated ([conventions](https://dplan-documentation.demos-europe.eu/development/guidelines-conventions/coding-styleguides/twig_html.html#guideline-for-naming-cypress-hooks))

### CODE REVIEW

**Review by breierDemos — CHANGES_REQUESTED** [review]
Code review only

File: demosplan/DemosPlanCoreBundle/ResourceTypes/GenericStatementAttachmentResourceType.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+ *
+ * @template-extends DplanResourceType<T>
+ *
+ * @property-read StatementResourceType $statement
+ * @property-read FileResourceType $file
+ */
+class GenericStatementAttachmentResourceType extends DplanResourceType
+{
+    public function __construct(
+        private readonly FileService $fileService, private readonly StatementRepository $statementRepository, private readonly FileContainerRepository $fileContainerRepository,
```
[the remark this answers is not in the document]
Comment (sierrazamudiodemos) [reply demosplan/DemosPlanCoreBundle/ResourceTypes/GenericStatementAttachmentResourceType.php]: @breierDemos removed: 4b9d7b3e3b656df5c751aed8489b44781cf1ac12

File: demosplan/DemosPlanCoreBundle/ResourceTypes/AbstractStatementResourceType.php [hunk 2]
```diff
@@ -99,12 +99,11 @@
  * @property-read End $isManual
  * @property-read End $manual
  * @property-read End $anonymous
- * @property-read FileResourceType $files @deprecated Use {@link StatementResourceType::$attachments} instead (needs implementation changes)
+ * @property-read FileResourceType $files @deprecated Use {@link StatementResourceType::sourceAttachments or @see StatementResourceType::genericAttachments} instead (needs implementation changes)
```
[the remark this answers is not in the document]
Comment (sierrazamudiodemos) [reply demosplan/DemosPlanCoreBundle/ResourceTypes/AbstractStatementResourceType.php]: @breierDemos done 958f2b38fb3df4461db3668ffa8fefb37b5f65f0

File: demosplan/DemosPlanCoreBundle/ResourceTypes/OriginalStatementResourceType.php [hunk 3]
```diff
@@ -143,7 +143,7 @@ protected function getProperties(): ResourceConfigBuilderInterface
             static fn (Statement $statement): bool => $statement->isTextPassagesAnonymized()
         );
         $originalStatementConfig->attachments
-            ->setRelationshipType($this->resourceTypeStore->getStatementAttachmentResourceType())
+            ->setRelationshipType($this->resourceTypeStore->getSourceStatementAttachmentResourceType())
```
[the remark this answers is not in the document]
Comment (sierrazamudiodemos) [reply demosplan/DemosPlanCoreBundle/ResourceTypes/OriginalStatementResourceType.php:146]: @breierDemos good. done a54d79000d0753f711716a81acad61fd1d650044

**Review by breierDemos — APPROVED** [review]
lgtm code review only

**Pull request merged**